### PR TITLE
Fix a bug with cyton daisy wifi

### PR DESCRIPTION
**Pull request by daniellasry**
Fix for OpenBCI/OpenBCI_GUI#837

The way we were doing cyton wifi daisy was wrong

- we were resetting the package array in each while loop, but it is suppoosed to persist so we can fill cyton samples then daisy samples
- cyton wifi daisy is different from serial daisy in that sample IDs are repeated (0, 0, 1, 1, 2, 2, 3, 3...) while serial daisy is not (0, 1, 2, 3, 4). So using even v.s. odd to distinguish cyton v.s. daisy was not working.
- The logic was broken with the first_sample stuff... we were actually treating it as "second sample"

Fixed up cyton daisy serial too to be consistent.

I foudn out that in cyton wifi daisy, the first sample is cyton and the second one is daisy. In cyton serial daisy, it's the opposite.

### CODE REVIEW

File: src/board_controller/openbci/cyton_daisy_wifi.cpp [hunk 1]
```diff
@@ -42,7 +42,10 @@ void CytonDaisyWifi::read_thread ()
     */
     int res;
     unsigned char b[OpenBCIWifiShieldBoard::package_size];
+    double package[30] = {0.};
+    bool first_sample = true;
     double accel[3] = {0.};
+    unsigned char last_sample_id = 0;
```
Comment (Andrey1994) [inline src/board_controller/openbci/cyton_daisy_wifi.cpp:48]: one question I have. Does board reset this index if streaming was stopped and restarted after that?

Comment (daniellasry) [reply src/board_controller/openbci/cyton_daisy_wifi.cpp:48]: It does not reset, but this code can handle both cases gracefully.

**Comment by Andrey1994** [comment]
I think it can not handle a case if board restarted streaming and didnt reset counter

**Comment by Andrey1994** [comment]
We will create a new thread and set last package id back to 0, board will send first package in new streaming session with random id and check last_id == byte[0] will not work as it should

**Comment by Andrey1994** [comment]
Oh, no seems like this case should work fine. But then I dont get how case with first package == 0 works.... initial value you use for last package id is 0. If board sends first package with id == 0 this code will work like its not a first package. Should initial value for last id be -1?

**Comment by daniellasry** [comment]
It will work with zero too, because first_sample's initial value is "true".
We can set it to -1 if it gives more peace of mind (it would have to be int instead of uchar)

What this code *really* needs is to be broken up and unit tested 😃

**Pull request merged**